core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorAccessControl.java

```diff
@@ -695,4 +696,20 @@ default Optional<ViewExpression> getColumnMask(ConnectorSecurityContext context,
     {
         return Optional.empty();
     }
+
+    /**
+     * Bulk method for getting column masks for a subset of columns in a table.
+     * <p>
+     * Constraints from {@link ConnectorAccessControl#getColumnMask(ConnectorSecurityContext, SchemaTableName, String, Type)}
+     * carry over to each mask returned.
+     *
+     * @return a list, each element a mask if present or empty, maintaining the order of elements
+     */
+    default Map<ColumnSchema, ViewExpression> getTableColumnMasks(ConnectorSecurityContext context, SchemaTableName tableName, List<ColumnSchema> columns)
```

Comment:
Deprecate the `getColumnMask`. We do not want the implementors to have to implement two methods, we will eventually remove the old one.